File: java/java-api/src/main/java/com/spring/api/board/service/BoardService.java
package com.spring.api.board.service;

import com.spring.api.board.dto.request.BoardCreateRequest;
import com.spring.api.board.dto.request.BoardUpdateRequest;
import com.spring.jpa.mysql.domain.Board;

import java.util.List;

public interface BoardService {
    // READ
    List<Board> getList();
    Board get(Long id);

    // CREATE
    Board create(BoardCreateRequest request);

    // UPDATE
    Board update(BoardUpdateRequest request);

    // DELETE
    void delete(Long id);
}

File: kotlin/kotlin-api/src/main/kotlin/com/spring/api/board/service/BoardService.kt
package com.spring.api.board.service

import com.spring.api.board.dto.request.BoardCreateRequest
import com.spring.api.board.dto.request.BoardUpdateRequest
import com.spring.jpa.mysql.domain.Board

interface BoardService {
    // READ
    fun getList(): List<Board>
    fun get(id: Long): Board

    // CREATE
    fun create(request: BoardCreateRequest): Board

    // UPDATE
    fun update(request: BoardUpdateRequest): Board

    // DELETE
    fun delete(id: Long)
}